zerver/tests/test_upload.py

```diff
@@ -38,12 +48,122 @@ def test_no_file_upload_failure(self):
+            user_profile = get_user_profile_by_email(sender_email)
+            two_week_ago = timezone.now() - datetime.timedelta(weeks=2)
+            dummy_1 = Attachment.objects.create(
+                                        file_name='zulip.txt',
+                                        path_id='1/31/4CBjtTLYZhk66pZrF8hnYGwc/zulip.txt',
+                                        owner=user_profile,
+                                        create_time=two_week_ago
+                                        )
+            dummy_2 = Attachment.objects.create(
+                                        file_name='temp_file.py',
+                                        path_id='1/31/4CBjtTLYZhk66pZrF8hnYGwc/temp_file.py',
+                                        owner=user_profile,
+                                        create_time=two_week_ago
+                                        )
+
+            #send message refering only dummy_1
+            self.subscribe_to_stream(sender_email, "Denmark")
+            body = "Some files here ...[zulip.txt](http://localhost:9991/user_uploads/1/31/4CBjtTLYZhk66pZrF8hnYGwc/zulip.txt)"
+            self.send_message(sender_email, "Denmark", Recipient.STREAM, body, "test")
+
+            #get list of old unclaimed atachments
+            weeks_ago = 2
+            old_attachments = get_old_unclaimed_attachments(weeks_ago)
+            self.assertIn(dummy_2, old_attachments)
+            self.assertTrue(dummy_1 not in old_attachments)
+
+class LocalStorageTest(AuthedTestCase):
+    test_files = []
+
+    #This test will go through the code path for uploading files onto LOCAL storage.
+    #So it is necessary that /home/zulip folder exists.
+    def test_file_upload_authed(self):
+        """
+        A call to /json/upload_file should return a uri and actually create an
+        entry in the database. This entry will be marked unclaimed till a message
+        refers it.
+        """
+        self.login("hamlet@zulip.com")
+        fp = StringIO("zulip!")
+        fp.name = "zulip.txt"
+
+        result = self.client.post("/json/upload_file", {'file': fp})
+        self.assert_json_success(result)
+        json = ujson.loads(result.content)
+        self.assertIn("uri", json)
+        uri = json["uri"]
+        base = '/user_uploads/'
+        self.assertEquals(base, uri[:len(base)])
+
+        #In the future, local file requests will follow the same style as S3
+        #requests; they will be first authenthicated and redirected
+        #response = self.client.get(uri)
+        #redirect_url = response['Location']
+
+        res = self.client.get(uri)
+        #check if the file recieved is correct
+        file_content = [i for i in res.streaming_content]
+        self.assertIn("zulip!", file_content)
+
+        #check if DB has attachment marked as unclaimed
+        entry = Attachment.objects.get(file_name='zulip.txt')
+        self.assertEquals(entry.is_claimed(), False)
```

Comment:
The first half of the code for this test is now a duplicate of test_file_upload_authed in FileUploadTest; can you merge these?